modules/@apostrophecms/piece-type/index.js

```diff
@@ -1053,6 +1053,9 @@ module.exports = {
           await self.apos.doc.db.deleteMany({ _id: { $in: deletes } });
           deletes.splice(0);
         }
+      },
+      checkBatchOperationsPermissions(req, permission = 'edit') {
+        return self.batchOperations.filter(batchOperation => self.apos.permission.can(req, permission, self.name, batchOperation.action));
```

Comment:
The `permission` to be checked should be coming from `batchOperation.permission`, if present, otherwise that paticular operation should always be allowed. The fourth argument (`mode`) should not be needed here, because the manage view is only accessible in draft mode.

Reply:
```
      can(req, action, docOrType, mode)
```